### ISSUE TITLE
Building Shell Tools

### ISSUE THREAD

**Issue by Sequoia** [nodeschool/workshoppers#31]
* arg parsing & libs for this
* switches aliases etc.
* designing to work with pipes
* sub-commands
* packaging for global install
* output stuff: colors, tables, progress bars
* taking input

**Comment by martinheidegger** [comment]
:+1:

**Comment by grabbou** [comment]
https://github.com/SBoudrias/Inquirer.js/ This might be also useful to cover but actually might be a part of completely another workshop.. e.g "Building interactive shell applications". Can cover this chapter just in case you think it's worth including.

**Comment by Sequoia** [comment]
ty @grabbou I will check it out!

**Comment by Sequoia** [comment]
@grabbou what do you mean about "can cover this chapter"? What chapter? Sorry I don't follow! :smile_cat:

**Comment by Sequoia** [comment]
@grabbou Thanks that lib looks awesome!! I had been using https://github.com/flatiron/prompt which appears to be completely derelict & afaict it's popularity rests on the fact that it took the name "prompt" in npm before anyone else could so it's "NPM SEO" is really good. Glad to have found its successor, or one of them!

**Comment by grabbou** [comment]
Sorry, I think I've eaten out "I" from that sentence :D

**Comment by Sequoia** [comment]
@grabbou oh cool thanks!! Let me get the ball rolling first, I have a *fairly* clear plan of how I want the workshopper to go... I'll share the google doc at you & you can add comments/edit suggestions there if you're interested!

**Comment by grabbou** [comment]
Yeah, cool. I've started doing my own recently and never expected it to be such a fun! Definitely, you can share the docs either here or via email and if only you want - you can count me in!

**Comment by martinheidegger** [comment]
Missing topics:
- Shell auto-complete
- Man pages

**Comment by Sequoia** [comment]
Heyyy @grabbou... old pal! I love your avi by the way :wink: 

j/k but in seriousness I'd totally love to take you up on your offer of help for CLIJS! I have sort of an aggressive timeline for a first release (about April 10) & I'm a little more than 1/2 way there. Not all the lessons I have slated are totally necessary but I have an idea for a core set.

See the slated lessons here: https://github.com/Sequoia/clijs/labels/Lesson

If you have ideas for new ones, add a ticket for discussion! If you can knock out one of those tickets (especially those that say **help wanted** especially especially those that say **help wanted** *and* **priority**) please go for it!

TIA & I look forward to collaborating with you and whoever else is willing & able!

**Comment by grabbou** [comment]
Hey haha, thanks. checkered shirt for the win :+1: 

Been a crazy week for me recently but I am available to release few lessons for your workshop! (After we're done, don't forget to help me with my EventEmitter one I am still somewhere at the beginning!)

Will check out the repo later today and let you know what I think and what lessons I can collaborate on.

**Comment by Sequoia** [comment]
@grabbou yes! totally will help with your workshop. thx again!! :tada:

**Comment by grabbou** [comment]
Thanks :) Crazy days as I am setting up the chapter in my city now, but will get back to you within couple of days.
